app/invocation/invocation.tsx

```diff
@@ -262,16 +263,19 @@ export default class InvocationComponent extends React.Component<Props, State> {
   }
 
   scheduleRefetch() {
-    clearTimeout(this.timeoutRef);
-    // Refetch invocation data in 3 seconds to update status.
-    this.timeoutRef = window.setTimeout(async () => {
-      // Before fetching the invocation, wait for the runner execution to be
-      // fetched, so we don't keep canceling the execution fetch if it takes
-      // longer than the invocation poll interval.
-      if (this.runnerExecutionRPC) await this.runnerExecutionRPC;
-
-      this.fetchInvocation();
-    }, 3000);
+    if (!this.refetchInProgress) {
+      this.refetchInProgress = true;
+      // Refetch invocation data in 3 seconds to update status.
+      this.timeoutRef = window.setTimeout(async () => {
+        // Before fetching the invocation, wait for the runner execution to be
+        // fetched, so we don't keep canceling the execution fetch if it takes
+        // longer than the invocation poll interval.
+        if (this.runnerExecutionRPC) await this.runnerExecutionRPC;
+
+        this.fetchInvocation();
+        this.refetchInProgress = false;
```

Comment:
sort of on the flip side of this issue--it seems like we've always allowed multiple fetchInvocation() calls to spawn at the same time (if one takes a long time).  maybe we should be waiting until the invocation fetch is done before we set `refetchInProgress = false`?  i might be missing some quirky edge case where we're genuinely relying on this to force-update the UI, but the change in this PR would already break that (if it exists).

Reply:
I think I addressed this by adding `await` - could you take a look and see if that's what you meant?

Reply:
@maggie-lou The `await` here doesn't actually await anything because `fetchInvocation()` doesn't return a promise

Reply:
I'm going to be honest - I have a loose understanding of how this works, but my understanding is that the return type is inferred and `fetchInvocation` implicitly returns a promise

Reply:
`fetchInvocation` would implicitly return a promise if it were marked `async` but because it's not marked async, there's no implicit transform applied to the return value. In this case it's always returning `undefined` which is the default return value when nothing gets returned (i.e. no `return` statement)

Reply:
Would adding `async` fix this?

Reply:
https://github.com/buildbuddy-io/buildbuddy/pull/7430

I verified that the previous code didn't work, and that this new version does work, with some logging and a sleep